### PR TITLE
add hydra links to GitHub comments

### PR DESCRIPTION
**Pull request by ryantm**
If a build is attempted, this tries to construct a valid link to the Hydra builds for the attrs. Sometimes it will fail because Hydra won't have a job for it.

### CODE REVIEW

**Comment by ryantm** [comment]
The test that is failing on Travis `tests::log_message_collector::test_logs_collect` is passing on my machine.

**Comment by LnL7** [comment]
Maybe there's a race condition in the test, can't  reproduce it either and the second build succeeded.

File: ofborg/src/tasks/githubcommentposter.rs [hunk 1]
```diff
@@ -99,7 +99,17 @@ fn result_to_comment(result: &BuildResult) -> String {
     reply.push("".to_owned());
 
     if let Some(ref attempted) = result.attempted_attrs {
-        reply.extend(list_segment("Attempted", attempted.clone()));
+        let target_branch = result.pr.clone().target_branch.unwrap_or("master".to_string());
+        reply.extend(list_segment(
+            "Attempted",
+            attempted.iter().map( |attempt| {
+                format!(
+                    "[{}](https://hydra.nixos.org/job/nixpkgs/{}/{}.{})",
```
Comment (LnL7) [inline ofborg/src/tasks/githubcommentposter.rs:115]: Using the branch isn't a good idea, none of the jobsets have the same name except for staging.

**Comment by ryantm** [comment]
@LnL7 Okay, I think I fixed the issue of branch to jobset correspondence.

**Comment by ryantm** [comment]
```
07:53 < ryantm> gchristensen: What do you think of https://github.com/NixOS/ofborg/pull/191 ?
07:53 < {^_^}> #191 (by ryantm, open): add hydra links to GitHub comments
07:55 < gchristensen> I don't like it much, because of the chances of 404s
07:55 < gchristensen> it is possible to be a not very strong signal
07:55 < gchristensen> I think it would be nicer to build before and after the commit
08:01 < gchristensen> because, then, too, much better signal can be provided: specifics on how the closure size has changed, if it could build before / could build now, build time
                      differences, build closure changes, etc
08:04 < ryantm> gchristensen: You're saying even if it doesn't 404 it might not be a good signal, right?
08:04 < gchristensen> yeah
08:04 < gchristensen> because then you have to know what you're looking for and how to find it, and then to know what it means
08:07 < ryantm> gchristensen: well, building before and after sounds great, and build caching might mean it's often pretty quick.
08:08 < ryantm> gchristensen: I'll go ahead and close that PR.
08:08 < gchristensen> true
08:08 < gchristensen> maybe we can fetch data from the Hydra API (or add data to the Hydra API so it is useful)
```

**Pull request closed without merging**